### DISCUSSION TITLE
Iroh on embedded devices?

### DISCUSSION THREAD

**Discussion by abhishekdhamale** [n0-computer/iroh#680] [General]
Hi, we are exploring Iroh for distribution of files on embedded devices. I would like to know about Iroh on embedded devices?

**Comment by flub** [comment]
Hello, I guess your entry point would be to see where using the iroh-embed crate gets you.  You'll probably have to tune a lot of things depending on just how many resources you have on the devices.

Iroh has so far not been particularly tuned for this, it does rely on a lot of high-level system interfaces and is probably impossible to run on no-std right now.  So depending on what kind of embedded devices you are considering will affect how feasible this is.  We'd be curious to hear about requirements and how you think ipfs should behave on those devices, but that would be longer-term planning is my guess.

**Reply by rklaehn** [reply]
What kind of embedded devices do you mean? There is a very large spectrum, from fully featured linux computers to very low powered ones.

In the medium term we definitely want to target embedded linux devices such as industrial pi computers such as this: https://revolutionpi.com/ .

However, a very low footprint embedded version will not be kubo compatible. See https://n0.computer/blog/a-new-direction-for-iroh/ .
